{{cookiecutter.project_slug}}/frontend/src/store/modules/main.ts

```diff
@@ -0,0 +1,252 @@
+    this.logInError = payload;
+  }
+
+  @Mutation
+  setUserProfile(payload: IUserProfile) {
+    this.userProfile = payload;
+  }
+
+  @Mutation
+  setDashboardMiniDrawer(payload: boolean) {
+    this.dashboardMiniDrawer = payload;
+  }
+
+  @Mutation
+  setDashboardShowDrawer(payload: boolean) {
+    this.dashboardShowDrawer = payload;
+  }
+
+  @Mutation
+  addNotification(payload: IAppNotification) {
+    this.notifications.push(payload);
+  }
+
+  @Mutation
+  removeNotification(payload: IAppNotification) {
+    this.notifications = this.notifications.filter(
+      (notification) => notification !== payload,
+    );
+  }
+
+  @Action
+  async logIn(payload: { username: string; password: string }) {
+    try {
+      const response = await api.logInGetToken(payload.username, payload.password);
+      const token = response.data.access_token;
+      if (token) {
+        saveLocalToken(token);
+        this.setToken(token);
+        this.setLoggedIn(true);
+        this.setLogInError(false);
+        await this.getUserProfile();
+        await this.routeLoggedIn();
+        this.addNotification({ content: "Logged in", color: "success" });
+      } else {
+        await this.logOut();
+      }
+    } catch (err) {
+      this.setLogInError(true);
+      await this.logOut();
+    }
+  }
+
+  @Action
+  async getUserProfile() {
+    try {
+      const response = await api.getMe(this.token);
+      if (response.data) {
+        this.setUserProfile(response.data);
+      }
+    } catch (error) {
+      await this.checkApiError(error);
+    }
+  }
+
+  @Action
+  async updateUserProfile(payload: IUserProfileUpdate) {
+    try {
+      const loadingNotification = { content: "saving", showProgress: true };
+      this.addNotification(loadingNotification);
+      const response = (
+        await Promise.all([
+          api.updateMe(this.token, payload),
+          await new Promise((resolve, _reject) => setTimeout(() => resolve(), 500)),
+        ])
+      )[0];
+      this.setUserProfile(response.data);
+      this.removeNotification(loadingNotification);
+      this.addNotification({
+        content: "Profile successfully updated",
+        color: "success",
+      });
+    } catch (error) {
+      await this.checkApiError(error);
+    }
+  }
+
+  @Action
+  async checkLoggedIn() {
+    if (!this.isLoggedIn) {
+      let token = this.token;
+      if (!token) {
+        const localToken = getLocalToken();
+        if (localToken) {
+          this.setToken(localToken);
+          token = localToken;
+        }
+      }
+      if (token) {
+        try {
+          const response = await api.getMe(token);
+          this.setLoggedIn(true);
+          this.setUserProfile(response.data);
+        } catch (error) {
+          await this.removeLogIn();
+        }
+      } else {
+        await this.removeLogIn();
+      }
+    }
+  }
+
+  @Action
+  async removeLogIn() {
+    removeLocalToken();
+    this.setToken("");
+    this.setLoggedIn(false);
```

Comment:
Adding `this.setUserProfile(null)` would fix the Bug mentioned above. Consider setting `setUserProfile(palyload: IUserProfile | null)` so no warning is printed.

Reply:
```suggestion
    this.setLoggedIn(false);
    this.setUserProfile(null);
```